question/question.c

```diff
@@ -185,6 +185,26 @@ int mutt_multi_choice(const char *prompt, const char *letters)
   return choice;
 }
 
+const int
+utf_length_table[256] = {
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 32 */
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 64 */ 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 96 */
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 128 */ 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 160 */
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,		/* 192 */ 
+	2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 
+	2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,		/* 224 */
+	3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 
+	4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 1, 1,		/* 256 */ 
+};
+
```

Comment:
Even though this is only used once, I suggest we:
- move this into `mutt/mbyte.c`
- make it static
- create a helper function to get the value

We use [`clang-format`](https://neomutt.org/dev/analysis/clang-format) on the code.
It's not kind to large tables, so please add these around the data:
```
// clang-format off
...
// clang-format on
```

Also, could we make the table `char[]` -- that would reduce the size by a factor of 4!

Reply:
Looking at `mutt/mbyte.c`, I think that the purpose of the table is equivalent to one of the purposes of the function `mbrtowc` used in `mutt_mb_charlen` to get the value `k` which is returned by `mutt_mb_charlen`, with the difference that only the first character is needed to determine the length.

Reply:
Based on the above however, I can add a separate function `mutt_mb_charlen_first` (or similarly named...), which would be used for this specific purpose.

Reply:
Thanks for adding the function.
Yes, it's a bit of overkill, but it's good encapsulation.